typed-builder-macro/src/field_info.rs

```diff
@@ -116,6 +121,7 @@ pub struct FieldBuilderAttr<'a> {
     pub default: Option<syn::Expr>,
     pub deprecated: Option<&'a syn::Attribute>,
     pub setter: SetterSettings,
+    pub implementations: Option<Vec<Vec<sus_impls::Dependency>>>,
```

Comment:
Also - does it have to be an `Option`? If no `requires` setting it given, why not generate a single implementation where all the fields are `Generic` expect for the current fields which would be `Unset`?

Reply:
I did indeed use the type `Implementation` but I did not push the changes yet.

I wanted to avoid allocating vectors in case that the `requires` attribute was not used. But I will find out if I should accept that tradeoff to simplify generating methods when I do use `implementations`.